File: backend/Models/Seller.cs
using System;
using System.Collections.Generic;

namespace backend.Models;

public partial class Seller
{
    public string Id { get; set; } = null!;

    public string UserId { get; set; } = null!;

    public string Slug { get; set; } = null!;

    public string DisplayName { get; set; } = null!;

    public string ImageUrl { get; set; } = null!;

    public bool IsVerified { get; set; }
    public bool IsClosed { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual ICollection<Offer> Offers { get; set; } = new List<Offer>();

    public virtual User User { get; set; } = null!;

    public Seller NormalizeForJson()
    {
        Offers = null!;
        User = null!;

        return this;
    }
}

File: frontend/src/lib/query/seller.ts
import { method } from "../fetch";
import { makeQuery } from "./util";

export type Seller = {
  id: string;
  slug: string;
  displayName: string;
  imageUrl: string;
  isVerified: boolean;
  isClosed: boolean;
  createdAt: string;
  updatedAt: string;
};

export function sellerMeQuery() {
  return makeQuery(
    ["seller", "me"],
    `${import.meta.env.VITE_BACKEND_PROD_URL}/api/seller/me`
  );
}

export function sellerByUserIdQuery(userId: string) {
  return makeQuery(
    ["seller", "user-id", userId],
    `${import.meta.env.VITE_BACKEND_PROD_URL}/api/seller/user-id/${userId}`
  );
}

export function sellerIdQuery(id: string) {
  return makeQuery(
    ["seller", "id", id],
    `${import.meta.env.VITE_BACKEND_PROD_URL}/api/seller/id/${id}`
  );
}

export function sellerSlugQuery(slug: string) {
  return makeQuery(
    ["seller", "slug", slug],
    `${import.meta.env.VITE_BACKEND_PROD_URL}/api/seller/slug/${slug}`
  );
}

export async function closeSellerAccount() {
  const response = await method(
    `${import.meta.env.VITE_BACKEND_PROD_URL}/api/seller/close-account`,
    "POST"
  );

  return response.result === "success";
}

export async function setSellerDisplayName(displayName: string) {
  const response = await method(
    `${import.meta.env.VITE_BACKEND_PROD_URL}/api/seller/display-name`,
    "POST",
    { displayName }
  );

  return response.result === "success";
}

export async function setSellerImageUrl(imageUrl: string) {
  const response = await method(
    `${import.meta.env.VITE_BACKEND_PROD_URL}/api/seller/image-url`,
    "POST",
    { imageUrl }
  );

  return response.result === "success";
}

export async function createSellerProfile(userId: string) {
  const response = await method(
    `${import.meta.env.VITE_BACKEND_PROD_URL}/api/seller`,
    "POST",
    { userId }
  );

  return response.result === "success";
}
